### DISCUSSION TITLE
Revolve 360 degrees?

### DISCUSSION THREAD

**Discussion by bullgom** [nschloe/pygmsh#497] [Q&A] [answered]
I tried to revolve a line 360 degrees using
`geometry.revolve(line, [*axis_line.start], [*axis_line.end], 6.28)`
Where both `line` and `axis_line` are created using `geometry.add_line`.
This throws an assertion that `degree < pi`. 
Why not `2*pi`? How can I revolve 360 degrees?

I've also tried `-6.28` which didn't throw an error, but no rotation at all.

**Comment by bullgom** [comment, marked as answer]
I used method in [_add_pipe_by_rectangle_rotation implementation](https://github.com/nschloe/pygmsh/blob/975f82ad5e08d08567f172fcc90352694cafe557/src/pygmsh/geo/geometry.py#L540)

I am getting an error using above method but, I think its somethings else's problem. If you want 360 degree revolve, have a look.

**Comment by nschloe** [comment]
> Why not 2*pi? How can I revolve 360 degrees?

That's a known shortcoming of gmsh. You can rotate in three steps of pi/3, by using the fact that `revolve()` gives you three return entities
```
top, extruded, lateral
```
You can extrude `top` again.
